src/megachatapi_impl.cpp

```diff
@@ -692,6 +692,62 @@ void MegaChatApiImpl::sendPendingRequests()
             });
             break;
         }
+        case MegaChatRequest::TYPE_PREVIEW_CHAT_LINK:
+        {
+            const char *link = request->getText();
+            string parsedLink = link;
+
+            size_t pos = parsedLink.find("#chat_");
+            if (pos == string::npos)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
+            if (parsedLink.length() !=  pos + 6 + 8 + 25 + 1)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
+            pos += 7;
+            string phstr = parsedLink.substr(pos, 8);
+            MegaChatHandle ph = UNDEF;
+            Base64::atob(phstr.c_str(), (byte*)&ph, MegaClient::CHATLINKHANDLE);
+
+            pos += 8;
+            string key;
+            string keystr = parsedLink.substr(pos);
+            Base64::atob(keystr, key);
+
+            if (ISUNDEF(ph) || key.length() != 16)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
```

Comment:
I will add here a check to discard the chatroom is part of the list of chatrooms and/or already in preview mode

Reply:
We can actually return requested data if the chat is already opened for preview. And only fail if we already participate in the chatroom as a peer. Already changed that.